make_a_plea/management/commands/zendesk.py

```diff
@@ -0,0 +1,212 @@
+    u'priority',
+    u'problem_id',
+    u'recipient',
+    u'requester_id',
+    u'satisfaction_rating',
+    u'status',
+    u'submitter_id',
+    u'type',
+    u'url',
+    u'result_type',
+    u'allow_channelback',
+    u'assignee_id',
+    u'brand_id',
+]
+
+# Some have structure like dicts or lists that will be strigified
+STRUCTURED_FIELDS = [
+    u'tags',
+    u'external_id',
+    u'fields',
+    u'forum_topic_id',
+    u'followup_ids',
+    u'collaborator_ids',
+    u'custom_fields',
+    u'sharing_agreement_ids',
+    u'via',
+]
+
+# Some may contain various line endings that will be coerced to \n
+NEWLINE_FIELDS = [
+    "description",
+    "subject",
+    "raw_subject",
+]
+
+
+class Command(BaseCommand):
+
+    option_list = BaseCommand.option_list + (
+        make_option(
+            '--to-csv',
+            action='store_true',
+            dest='to-csv',
+            default=False,
+            help='Export all relevant zendesk items to csv file'),
+        )
+
+    def handle(self, *args, **options):
+        self.load_settings()
+        self.init_zen()
+
+        if options["to-csv"]:
+            self.to_csv()
+
+    def load_settings(self):
+        with open(os.path.expanduser("~/.hmcts-zendesk"), "r") as config_file:
+            self.settings = yaml.load(config_file.read())
+
+    def init_csv(self, f, dialect=csv.excel):
+        self.writer = csv.writer(f, dialect=dialect)
+
+    def init_zen(self):
+        self.zendesk = Zendesk(
+            self.settings["url"],
+            self.settings["email"],
+            self.settings["password"])
+
+    def get_all_fieldnames(self):
+        return \
+            SIMPLE_FIELDS + \
+            NEWLINE_FIELDS + \
+            STRUCTURED_FIELDS
+
+    def to_csv(self):
+        self.outfile = self.settings.get('outfile', 'zendump.csv')
+        self.batch_size = 100
+        with open(self.outfile, "wb") as f:
+
+            self.init_csv(f)
+            self.writerow(self.get_all_fieldnames())  # CSV Header
+
+            for batch in self.yield_ticket_batches():
+                for ticket in batch:
+                    self.writerow(self.ticket_to_row(ticket))
+
+    def writerow(self, row):
+        self.writer.writerow([
+            cell.encode("utf-8")
+            for cell in row])
+
+    def ticket_to_row(self, ticket):
+        return \
+            self.get_simple_fields(ticket) + \
+            self.get_newline_fields(ticket) + \
+            self.get_structured_fields(ticket)
+
+    def parse_simple_field(self, field):
+        try:
+            return field.encode("utf-8")
+        except AttributeError:  # ints
+            return str(field).encode("utf-8")
+
+    def get_simple_fields(self, ticket):
+        fields = []
+        for field in SIMPLE_FIELDS:
+            if field in ticket:
+                fields.append(self.parse_simple_field(ticket[field]))
+            else:
+                fields.append("")
+        return fields
```

Comment:
if `parse_simple_field` were updated to handle `None` then this function could become
```python
return [self.parse_simple_field(ticket.get(field)) for field in SIMPLE_FIELDS]
```